ldk-sample/src/main/java/com/getlipa/ldk/sample/LDKSample.java

```diff
@@ -0,0 +1,360 @@
+
+            @Override
+            public Option_C2Tuple_usizeTransactionZZ register_output(WatchedOutput output) {
+                relevantOutputs.add(output);
+                return Option_C2Tuple_usizeTransactionZZ.none();
+            }
+        });
+
+        // Step 8
+        final ChainMonitor chainMonitor = ChainMonitor.of(filter, txBroadcaster, logger, feeEstimator, persist);
+
+        // Step 9
+        final var seed = Hex.decode(SEED);
+        final var startupTime = System.currentTimeMillis();
+        final var keyManager = KeysManager.of(
+                seed,
+                startupTime / 1000,
+                (int) (startupTime * 1000)
+        );
+
+        // Step 10
+        final var directory = new File("data/channels");
+        var channelMonitors = new byte[][]{};
+        if (directory.exists() && directory.isDirectory()) {
+            channelMonitors = Arrays.stream(directory.listFiles())
+                    .map(File::toPath)
+                    .map(path -> {
+                        try {
+                            return (Files.readAllBytes(path));
+                        } catch (IOException e) {
+                            return null;
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    .toArray(byte[][]::new);
+        }
+
+        // Step 11
+        final var channelManagerFile = new File("data/channel-manager.dat");
+        byte[] serializedChannelManager = null;
+        if (channelManagerFile.exists() && channelManagerFile.isFile()) {
+            serializedChannelManager = Files.readAllBytes(channelManagerFile.toPath());
+        }
+
+        ChannelManagerConstructor channelManagerConstructor;
+        if (serializedChannelManager == null) {
+            final var currentBlockHeight = chainBackend.blockHeight(); //126;
+            final var latestBlockHash = chainBackend.blockHash(currentBlockHeight);
+            channelManagerConstructor = new ChannelManagerConstructor(
+                    Network.LDKNetwork_Regtest,
+                    UserConfig.with_default(),
+                    latestBlockHash,
+                    currentBlockHeight,
+                    keyManager.as_KeysInterface(),
+                    feeEstimator,
+                    chainMonitor,
+                    null,
+                    txBroadcaster,
+                    logger
+            );
+        } else {
+            channelManagerConstructor = new ChannelManagerConstructor(
+                    serializedChannelManager,
+                    channelMonitors,
+                    keyManager.as_KeysInterface(),
+                    feeEstimator,
+                    chainMonitor,
+                    filter,
+                    null,
+                    txBroadcaster,
+                    logger
+            );
+        }
+        final ChannelManager channelManager = channelManagerConstructor.channel_manager;
+
+        // Step 6
```

Comment:
Indeed, there's no required ordering except for the dependencies. Any suggestions for how we can better capture that in the docs?

Reply:
I guess most devs will read through the guide and do the implementations step by step. Therefore, it would be helpful if the ordering in the guide corresponds to the dependencies.